src/hdmf/container.py

```diff
@@ -855,8 +865,10 @@ def set_data_io(self, dataset_name: str, data_io_class: Type[DataIO], data_io_kw
             )
             data_io_kwargs = kwargs
         data = self.fields.get(dataset_name)
+        data_chunk_iterator_kwargs = data_chunk_iterator_kwargs or dict()
         if data is None:
             raise ValueError(f"{dataset_name} is None and cannot be wrapped in a DataIO class")
+        data = data_chunk_iterator_class(data=data, **data_chunk_iterator_kwargs)
```

Comment:
I just realized using `DataChunkIterator` as a default argument is not a good idea. In most cases a user will just want to set `DataIO` and not apply a `DataChunkIterator`, but the current behavior here is to always set a `DataChunkIterator`. I.e., I would suggest changing the default for the `data_chunk_iterator_class` to `None` and then here do 
```suggestion
        if data_chunk_iterator_class is not None:
            data = data_chunk_iterator_class(data=data, **data_chunk_iterator_kwargs)
```